### PR TITLE
1.6/1935 add a spinner to the trader page for large amounts of data vouchers displayed vvv ee

### PR DESCRIPTION
**Pull request by lauragurney**
https://trello.com/c/SP7VGSAQ/1935-add-a-spinner-to-the-trader-page-for-large-amounts-of-data-vouchers-displayed-vvv-ee

### CODE REVIEW

File: src/pages/Spinner.vue [hunk 1]
```diff
@@ -0,0 +1,74 @@
+<template id="spinner">
+    <div v-if="active" class="spinner-wrapper">
+        <div class="spinner">
+            <div></div>
```
Comment (charlesstrange2) [inline src/pages/Spinner.vue:4]: The use of these divs to create height is a bit of a sledgehammer;
wd cna probably solve that with css.

File: src/pages/Spinner.vue [hunk 2]
```diff
@@ -0,0 +1,74 @@
+<template id="spinner">
```
Comment (charlesstrange2) [inline src/pages/Spinner.vue:1]: Spinner.vue probably belongs under /components rather than /pages

File: src/pages/Account.vue [hunk 3]
```diff
@@ -10,190 +10,205 @@
                         v-bind:state="message.state"
                     ></message>
                 </transition>
+                <div v-if="spinnerActive">
+                    <spinner
+                        v-bind:active="spinnerActive"
+                        text="Please wait, working on it..."
+                    />
+                </div>
+                <div v-if="!spinnerActive">
```
Comment (charlesstrange2) [inline src/pages/Account.vue]: Could also be "v-else", which when applied to the next element after a "v-if", will act as an else for. It.

File: src/pages/Account.vue [hunk 4]
```diff
@@ -293,13 +308,21 @@ export default {
...
         },
         pgChangePage: function (event) {
+            this.loadSpinner();
             const key = event.target.id;
             const pg = this.voucherPagination || {};
             const page = pg.hasOwnProperty(key) ? pg[key].page : 1;
             Store.getVoucherPaymentState(page);
         },
+        loadSpinner: function () {
+            this.showSpinner();
```
Comment (charlesstrange2) [inline src/pages/Account.vue]: Currnently this looks like
- mount the page
- show the spinner
- set a timeout to hide th spinner at 2 aeconds
- fetch the data <- may take more than 2 seconds
- time out triggers
- data may not have arrived.

It's OK for now, but we should remember to tie spinner hiding to data-has-been-fetched when we fix that bit.

And I don't have a concrete idea about that at the moment.
I'm thinking
- Store has a flag that says it's fetching for us
- we watch that in data here
- when it changes to true, we start a timer to show spinner
- when it goes false, we cancel timer and hide spinner. 
- if it takes longer than timer, spinner shows
But that's for another card.

**Review by charlesstrange2 — CHANGES_REQUESTED** [review]
Try the v-else, cull some divs and move the vue file location

File: src/pages/Spinner.vue (hunk 1 above)
Comment (lauragurney) [reply src/pages/Spinner.vue:4]: Removing them means there is no space for the spinny circle so it just disappears entirely. I checked other sources and they all said 4 x divs to allow for the spinner.

File: src/pages/Spinner.vue (hunk 2 above)
Comment (lauragurney) [reply src/pages/Spinner.vue:1]: Good shout, done.

File: src/pages/Account.vue (hunk 3 above)
Comment (lauragurney) [reply src/pages/Account.vue]: Done

**Review by charlesstrange2 — APPROVED** [review]
♻️

**Pull request merged**